playground/backend/internal/preparators/java_preparators.go

```diff
@@ -0,0 +1,129 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package preparators
+
+import (
+	"bufio"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+// GetJavaPreparation return preparation methods that should be applied to Java code
+func GetJavaPreparation(filePath string) *[]Preparator {
+	preparatorArgs := make([]interface{}, 1)
+	preparatorArgs[0] = filePath
+
+	publicClassModification := Preparator{
+		Prepare: removePublicClassModifier,
+		Args:    preparatorArgs,
+	}
+	additionalPackage := Preparator{
+		Prepare: removeAdditionalPackage,
+		Args:    preparatorArgs,
+	}
+	preparation := []Preparator{publicClassModification, additionalPackage}
+	return &preparation
+}
+
+// removePublicClassModifier removes public modification for class from java file
+func removePublicClassModifier(args ...interface{}) error {
+	filePath := args[0].(string)
+	if err := replace(filePath, "public class ", "class "); err != nil {
+		log.Printf("Preparation: Error during remove public modification for class, err: %s\n", err.Error())
+		return err
+	}
+	return nil
+}
+
+// removeAdditionalPackage removes packages from java file
+func removeAdditionalPackage(args ...interface{}) error {
+	filePath := args[0].(string)
+	if err := replace(filePath, `package ([\w]+\.)+[\w]+;`, ""); err != nil {
+		log.Printf("Preparation: Error during remove additional package, err: %s\n", err.Error())
+		return err
+	}
+	return nil
+}
+
+// createTempFile creates temporary file near with originalFile
+func createTempFile(originalFilePath string) (*os.File, error) {
+	// all folders which are included in filePath
+	filePathSlice := strings.Split(originalFilePath, "/")
+	fileName := filePathSlice[len(filePathSlice)-1]
+
+	// find parent folder for file
+	folderPath := "/"
+	filePathSlice = filePathSlice[:len(filePathSlice)-1]
+	for _, folder := range filePathSlice {
+		if folder == "" {
+			continue
+		}
+		folderPath = filepath.Join(folderPath, folder)
+	}
+	return os.Create(folderPath + "/tmp_" + fileName)
```

Comment:
If it's const - it should be moved to const

Reply:
Changed.